boot/bootstate20.go

```diff
@@ -421,24 +437,24 @@ func (bs20 *bootState20Base) markSuccessful(update bootStateUpdate) (bootStateUp
 	return u20, nil
 }
 
-func (bs20 *bootState20Base) setNext(next snap.PlaceInfo) (rebootRequired bool, u bootStateUpdate, err error) {
+func (bs20 *bootState20Base) setNext(next snap.PlaceInfo) (rbi RebootInfo, u bootStateUpdate, err error) {
 	u20, nextStatus, err := genericSetNext(bs20, next)
 	if err != nil {
-		return false, nil, err
+		return RebootInfo{RebootRequired: false}, nil, err
 	}
 
 	// if we are setting a snap as a try snap, then we need to reboot
-	rebootRequired = false
+	rbi.RebootRequired = false
 	if nextStatus == TryStatus {
 		// only update the try base if we are actually in try status
 		u20.writeModeenv.TryBase = next.Filename()
-		rebootRequired = true
+		rbi.RebootRequired = true
```

Comment:
could use a comment like:
```
// a 'try' base is handled by snap-bootstrap, hence we are not interested
// in the bootloader's opinion
```

Reply:
Good point, I've added a couple of addditional comments.